### ISSUE TITLE
Neural network, visualize the output

### ISSUE THREAD

**Issue by geekco2000** [deeplearning4j/deeplearning4j#3207]
I know you are all actually working hard, but I have got a question, not really an issue, so forgive in advance if this can disturb !!
I have runned a regression problem, using MultiLayerNetwork in Spark (with SparkDl4jMultiLayer), trained my model, then when I test the model with a new dataset (testData), I have got some really good result (RMSE : 3.94229e-04). I runned differents models to be sure this metric work well, it seems to work but I need to be sure, for that I have looked for ways to print the output (y calculated by the model, then compare the Y calculated with the real Y:
val evaluation = sparkNet.evaluateRegression(testData)
println(evaluation.stats)
display Y ??
But finally, I haven't found something useful. So, Is there some ways to print the output (Y^) of the result or Do I need to program it ? 

If this can help to give me a clue (or for anybody else who'd like to have an example :)), the configuration : 
20 neurons in input 
1 neuron in output (Identity activation function / MSE metric) 
1 hidden layer (50 neurons, tanh activation function) 
1000 iterations (by default), none epoch 
dataset of 100 000 observations (20 features to approximate 1 variable)

**Comment by geekco2000** [comment]
Thank you by the way, sorry I forgot..

**Comment by tomthetrainer** [comment]
You may have to program that yourself, I will see what visualization tools we have available. 

If you need an example as a starting point here is a classification example that plots values after evaluation. 

I understand you problem is regression, but there still may be some crossover. 

https://github.com/deeplearning4j/dl4j-examples/blob/master/dl4j-examples/src/main/java/org/deeplearning4j/examples/feedforward/classification/MLPClassifierLinear.java

**Comment by geekco2000** [comment]
Thank you for your help !
But, I didn't really mean a visualization tools, but that would be the best anyway :) 
I mean just a function to get the data calculated by the model, like when you call this function : 
val evaluation = sparkNet.evaluateRegression(testData)
I'm pretty sure the data calculated by the model (sparkNet) are stored somewhere (y calculated and y real), which allow you to calculate the RMSE metric !
It means get the result, output (y^) from an input data (X) in the model. Because I get the error metric but I can't really see any result nowhere (output). 

Otherwise, I'm gonna try to program that myself, then share my code for anyone else. 

Thanks again.

**Comment by tomthetrainer** [comment]
To get the output of a model use the output method.

INDArray output = model.output(input);

**Comment by geekco2000** [comment]
Great, got it !

	val net = sparkNet.fit(trainData)
        val labels = test.getLabels
	println("*** LABELS ***")
	println(labels)
	val predicted = net.output(test.getFeatures)
	println("*** PREDICTED ***")
	println(predicted)

--> If someone needs help 

Thanks you very much giving me your time :)

Something else if I may say so,

When I am trying to print the output I got some errors (train set : 70 000, test set : 30 000): 
[  ?,  ?,  ?,  ?, ...] --> it doesn't print correctly (weird characters), looks like something is missing, but the rmse looks very good, what this means ? 
Otherwise, when I train my model (for example) in a train set of 15 observations and a test set of 5 observations, then I can see the output printed correctly, but the rmse isn't good at all (which is quite normal). 
	val predicted = net.output(test.getFeatures)
	println("*** PREDICTED ***")
	println(predicted)
        [12 367, 15 768, 11 676, ...] --> looks like this 

Hard to find where I got an error :\ 

Thanks again !

**Comment by tomthetrainer** [comment]
If you continue to have issues with the printable characters after you familiarize yourself the structure of the output and therefore you suspect a bug on our end please re-open an issue. I am closing this,  thanks for posting your question glad I could help.
